From d7488a7d1661f79742d71c5cd0f6039f214aca40 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 24 Apr 2024 12:29:22 +0200
Subject: [PATCH] Permit to run gz sim -g on Windows

Signed-off-by: Silvio Traversaro <silvio@traversaro.it>
---
 src/cmd/cmdsim.rb.in | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/cmd/cmdsim.rb.in b/src/cmd/cmdsim.rb.in
index 64ee638c7b..65946d3494 100755
--- a/src/cmd/cmdsim.rb.in
+++ b/src/cmd/cmdsim.rb.in
@@ -598,12 +598,6 @@ See https://github.com/gazebosim/gz-sim/issues/168 for more info."
             options['record-period'], options['seed'])
             # Otherwise run the gui
       else options['gui']
-        if plugin.end_with? ".dll"
-          puts "`gz sim` currently only works with the -s argument on Windows.
-See https://github.com/gazebosim/gz-sim/issues/168 for more info."
-          exit(-1)
-        end
-
         ENV['RMT_PORT'] = '1501'
         Importer.runGui(options['gui_config'], options['file'],
                         options['wait_gui'], options['render_engine_gui'],
