From 75658fbe39ad4bb58947e0791037f3f5c75bc4be Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sat, 11 Oct 2025 13:09:13 +0200
Subject: [PATCH] Disable Quick Start Dialog on Windows

Signed-off-by: Silvio Traversaro <silvio@traversaro.it>
---
 src/gui/Gui.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/gui/Gui.cc b/src/gui/Gui.cc
index 548c61da80..ecbeff016c 100644
--- a/src/gui/Gui.cc
+++ b/src/gui/Gui.cc
@@ -258,6 +258,13 @@ std::string launchQuickStart(int &_argc, char **_argv,
 
   gzdbg << "Reading Quick start menu config." << std::endl;
   auto showDialog = dialog->ReadConfigAttribute(_configInUse, "show_again");
+  // Quick start menu is not supported on Windows at the moment, see
+  // https://github.com/gazebosim/gz-sim/issues/3106
+  #ifdef _WIN32
+  gzdbg << "Hardcoding Quick start menu config to "
+        << "false as we are on Windows." << std::endl;
+  showDialog = "false";
+  #endif
   if (showDialog == "false")
   {
     gzmsg << "Not showing Quick start menu." << std::endl;
