diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 98bbe66650..06295abf3b 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -1,3 +1,18 @@
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  cmake_minimum_required(VERSION 3.16)
+  set(GZ_SIM_VER 10)
+  set(GZ_COMMON_VER 7)
+  project(gz-sim-python VERSION ${GZ_SIM_VER})
+  find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
+  find_package(pybind11 REQUIRED)
+  find_package(gz-sim REQUIRED)
+  set(PROJECT_LIBRARY_TARGET_NAME "gz-sim::gz-sim")
+  include(CTest)
+  if(BUILD_TESTING)
+    enable_testing()
+  endif()
+endif()
+
 if(USE_SYSTEM_PATHS_FOR_PYTHON_INSTALLATION)
   if(NOT Python3_SITEARCH)
     # Get variable from Python3 module
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9f309499c8..f3a82dcac0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -250,7 +250,7 @@ add_subdirectory(examples)
 #============================================================================
 gz_create_packages()
 
-if (pybind11_FOUND AND NOT SKIP_PYBIND11)
+if (FALSE)
 	add_subdirectory(python)
 endif()
 #============================================================================
