From c4df5676d4a07bd1ff27f2cc4f5838903378e844 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 11 Oct 2023 13:25:21 +0200
Subject: [PATCH] Support cross-compilation

Signed-off-by: Silvio Traversaro <silvio@traversaro.it>
---
 src/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6e666dafc9..0aad6089e1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,6 +10,10 @@ get_target_property(msgs_desc_file
   gz-msgs${GZ_MSGS_VER}::gz-msgs${GZ_MSGS_VER} GZ_MSGS_DESC_FILE)
 
 gz_msgs_generate_messages_impl(
+  PYTHON_INTERPRETER
+    ${gz-msgs_PYTHON_INTERPRETER}
+  PROTOC_EXEC
+    ${gz-msgs_PROTOC_EXECUTABLE}
   MSGS_GEN_SCRIPT
     ${gz-msgs10_PROTO_GENERATOR_SCRIPT}
   FACTORY_GEN_SCRIPT
